tapir/shifts/views/views.py

```diff
@@ -309,3 +314,210 @@ def get(self, request, *args, **kwargs):
         call_command("run_freeze_checks")
         messages.info(request, _("Frozen statuses updated."))
         return super().get(request, args, kwargs)
+
+
+class SolidarityShiftUsed(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        date = datetime.now()
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
+        solidarity_shift = SolidarityShift.objects.filter(is_used_up=False)[0]
+        solidarity_shift.is_used_up = True
+        solidarity_shift.date_used = date
+        solidarity_shift.save()
+
+        ShiftAccountEntry(
+            user=tapir_user,
+            value=1,
+            date=date,
+            description="Solidarity shift received",
+            is_from_welcome_session=False,
+        ).save()
+
+        messages.info(
+            request, _("Solidarity Shift received. Account Balance increased by 1.")
+        )
+
+        return redirect(tapir_user.get_absolute_url())
+
+
+class SolidarityShiftGiven(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
+        shift_attendance = tapir_user.shift_attendances.filter(
+            is_solidarity=False, state=2
+        )[0]
+
+        SolidarityShift.objects.create(shiftAttendance=shift_attendance)
+
+        shift_attendance.is_solidarity = True
+        shift_attendance.save()
+
+        ShiftAccountEntry(
+            user=tapir_user,
+            value=-1,
+            date=datetime.now(),
+            description="Solidarity shift given",
+            is_from_welcome_session=False,
+        ).save()
+
+        messages.info(
+            request, _("Solidarity Shift given. Account Balance debited with -1.")
+        )
+
+        return redirect(tapir_user.get_absolute_url())
+
+
+class SolidarityShiftsView(LoginRequiredMixin, TemplateView):
+    template_name = "shifts/solidarity_shifts_overview.html"
+    model = SolidarityShift
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["available_solidarity_shifts"] = SolidarityShift.objects.filter(
+            is_used_up=False
+        ).count()
+        context["used_solidarity_shifts_total"] = SolidarityShift.objects.filter(
+            is_used_up=True
+        ).count()
+        context["gifted_solidarity_shifts_total"] = SolidarityShift.objects.count()
+        return context
+
+
+class CacheMonthsFirstSolidarityToTodayMixin:
```

Comment:
I would work with `date` objects as much as possible, and format them at the last moment. 
You could return dates like in https://github.com/SuperCoopBerlin/tapir/blob/9624b238c378b73f00863832b3b1a4c5c8c9b72a/tapir/coop/views/statistics.py#L207-L223
and when you build the context_data, set the `"labels"` to `[date.strftime(...) for date in dates]`

This way you won't need to do the month map yourself. This is the reference for the formatting codes : https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes